lib/wlib/memory/Memory.cpp

```diff
@@ -238,3 +358,61 @@ extern "C" void *memory_realloc(void *oldMem, size_t size) {
         return nullptr;
     }
 }
+
+uint32_t getTotalMemoryUsed() {
+    uint16_t totalMemory = 0;
+
+    for (auto &_allocator : _allocators)
+        totalMemory += _allocator->GetNumAllocations() * _allocator->GetBlockSize();
+
+    return totalMemory;
+}
+
+
+uint32_t getTotalMemoryAvailable() {
+    uint16_t totalMemory = 0;
+
+    for (auto &_allocator : _allocators)
+        totalMemory += _allocator->GetTotalBlocks() * _allocator->GetBlockSize();
+
+    return totalMemory;
+}
+
+bool isSizeAvailable(uint16_t blockSize) {
+    for (auto &_allocator : _allocators)
+        if (_allocator->GetBlockSize() == blockSize)
+            return true;
+
+    return false;
+}
+
+bool isSizeMemAvailable(uint16_t blockSize) {
+    for (auto &_allocator : _allocators)
+        if (_allocator->GetBlockSize() == blockSize) {
+            uint32_t totalMemAvail = _allocator->GetTotalBlocks() * _allocator->GetBlockSize();
+            uint32_t totalMemUsed = _allocator->GetNumAllocations() * _allocator->GetBlockSize();
+            return static_cast<bool>(totalMemAvail - totalMemUsed);

```

Comment:
You should do `totalMemAvail - totalMemUsed != 0` instead of casting to `bool`

Reply:
yeah true will fix that now